arch/x86/soc/ia32/soc.c

```diff
@@ -17,3 +17,22 @@
 #include <uart.h>
 #include <device.h>
 #include <init.h>
+#include <mmustructs.h>
+
+#ifdef CONFIG_X86_MMU
+
+#ifdef CONFIG_XIP
+MMU_BOOT_REGION(CONFIG_PHYS_LOAD_ADDR, CONFIG_ROM_SIZE*1024, MMU_ENTRY_READ);
+
+MMU_BOOT_REGION(CONFIG_PHYS_RAM_ADDR, CONFIG_RAM_SIZE*1024, MMU_ENTRY_WRITE);
+#else
+MMU_BOOT_REGION(CONFIG_PHYS_LOAD_ADDR, CONFIG_RAM_SIZE*1024, MMU_ENTRY_WRITE);
+#endif /*CONFIG_XIP */
+
+MMU_BOOT_REGION(CONFIG_LOAPIC_BASE_ADDRESS, 4096*1024, MMU_ENTRY_WRITE);
+MMU_BOOT_REGION(CONFIG_IOAPIC_BASE_ADDRESS, 1024*1024, MMU_ENTRY_WRITE);
+
+/*HPET */
+MMU_BOOT_REGION(CONFIG_HPET_TIMER_BASE_ADDRESS, 1024*1024, MMU_ENTRY_WRITE);
```

Comment:
I couldn't find the start address of this region in that manual. Hence I thought that there maybe a different memory map. I will change it to 4KB.